src/WebJobs.Script/Host/HostIdValidator.cs

```diff
@@ -117,15 +120,21 @@ private void HandleCollision(string hostId)
             }
 
             string message = string.Format(Resources.HostIdCollisionFormat, hostId);
+
             if (level == LogLevel.Warning)
             {
                 _logger.LogWarning(message);
+
+                DiagnosticEventLoggerExtensions.LogDiagnosticEventInformation(_logger, HostIdCollisionErrorCode, message, HostIdCollisionHelpLink);
             }
             else
             {
-                // we only allow Warning/Error levels to be specified, so anything other than
-                // Warning is treated as Error
+                // We only allow Warning/Error levels to be specified,
+                // so anything other than Warning is treated as Error.
                 _logger.LogError(message);
+
+                DiagnosticEventLoggerExtensions.LogDiagnosticEventError(_logger, HostIdCollisionErrorCode, message, HostIdCollisionHelpLink, new Exception(message));
```

Comment:
Can we use a more specific exception here? We can either create a custom one (if one doesn't already exist) or use an existing exception type i.e. argument exception or invalid operation

https://www.tutorialsteacher.com/csharp/csharp-exception